intake/catalog/local.py

```diff
@@ -502,18 +502,15 @@ def register_plugin_module(mod):
             register_driver(k, v, clobber=True)
 
 
-def get_dir(path):
-    if "://" in path:
-        protocol, _ = split_protocol(path)
-        out = get_filesystem_class(protocol)._parent(path)
-        if "://" not in out:
-            # some FSs strip this, some do not
-            out = protocol + "://" + out
-        return out
-    path = make_path_posix(os.path.join(os.getcwd(), os.path.dirname(path)))
-    if path[-1] != "/":
-        path += "/"
-    return path
+def get_dir(path: str) -> str:
+    """Get the absolute path of the directory of the file"""
+    protocol, pure_path = split_protocol(path)
+    dirname = os.path.dirname(pure_path)
+    if protocol:
+        # leave the path without '/' in the end
+        return f"{protocol}://{dirname}"
+    # get absolute path, convert to posix and add '/' in the end
+    return f"{os.path.abspath(dirname).replace(os.sep, r'/')}/"
```

Comment:
Will this not produce the OS-dependent version of the path?
I wonder whether `fsspec.core.url_to_fs` might be a more appropriate function for this job.

Reply:
I think it would be quite challenging due to [its strange behavior](https://github.com/fsspec/filesystem_spec/discussions/1356).
I have prepared another solution #768